language/vm/vm_runtime/src/block_processor.rs

```diff
@@ -84,6 +87,9 @@ pub fn execute_block<'alloc>(
         };
         report_execution_status(output.status());
         data_cache.push_write_set(&output.write_set());
+
+        // `result` is initally empty and not added to before this point
+        assume!(result.len() < usize::max_value());
```

Comment:
I am not reassured. We are inside a loop. Why are we sure result cannot overflow? The fact that result starts out empty does not assure this.

I think we might argue that every transaction pays gas proportional to its output and that gas limits will ensure that we never get anywhere close to usize::max_value().

Reply:
Updated the comment to make it more clear what's going on here.